pkg/cli/config/config.go

```diff
@@ -18,10 +18,11 @@ import (
 )
 
 const (
-	timeout     = 15 * time.Second
-	openAPIPath = "/openapi"
-	apiVersion  = "v1"
-	versionPath = "/debug/version"
+	shortTimeout = 1 * time.Second
+	timeout      = 5 * time.Second
```

Comment:
@aiwantaozi  the `client.Timeout` controls the request from dial to response header reading, I think 1s is too short for some bad networking connection cases.

we should configure more granular timeouts, here is an example.
```
&http.Client{
    Transport: &http.Transport{
        Proxy: http.ProxyFromEnvironment,
        DialContext: (&net.Dialer{
            Timeout:   1 * time.Second,           // shorten this to make sure if we can connect to server
            KeepAlive: 30 * time.Second,  
            DualStack: true,
        }).DialContext,
        TLSHandshakeTimeout:   10 * time.Second,  // server must finish cert generation if using self-signed CA
        ResponseHeaderTimeout: 30 * time.Second,  // server must complete the request and start to transit the response 
    }
}
```

Reply:
updated